cls_module/cls_module/memory/stm/aha.py

```diff
@@ -315,23 +402,60 @@ def forward_memory(self, inputs, targets, labels):
 
     losses = {}
     outputs = {}
+    features = {}
 
     outputs['ps'] = self.forward_ps(inputs)
+    features['ps'] = outputs['ps'].detach().cpu()
 
     # Compute DG Overlap
     overlap = self.ps.compute_overlap(outputs['ps'])
     losses['ps_overlap'] = overlap.sum()
 
-    pr_targets = outputs['ps'] if self.training else self.pc_buffer_batch
-    losses['pr'], outputs['pr'] = self.forward_pr(inputs=inputs, targets=pr_targets)
+    # Perforant Pathway: Hebbian Learning
+    if self.is_hebbian_perforant():
+      with torch.no_grad():
+        dg_ca3_in = outputs['ps']
+        pre_dg_ca3_out = self.dg_ca3(dg_ca3_in)
 
-    # Compute PR Mismatch
-    pr_out = outputs['pr']['pr_out']
-    pr_batch_size = pr_out.shape[0]
-    losses['pr_mismatch'] = torch.sum(torch.abs(pr_targets - pr_out)) / pr_batch_size
+        ec_ca3_in = torch.flatten(inputs, 1)
+        pre_ec_ca3_out = self.ec_ca3(ec_ca3_in)
 
-    pc_cue = outputs['ps'] if self.training else outputs['pr']['z_cue']
-    outputs['pc'] = self.forward_pc(inputs=pc_cue)
+        pre_pc_out = pre_dg_ca3_out + pre_ec_ca3_out
```

Comment:
Is this a confusing variable name?
i.e. Is this the input to PC (including contributions from dg and from ec)

Reply:
This was resolved. Renamed to `pc_cue` and restored the `pc_buffer` functionality in the hebbian pathway